python/rmm/docs/conf.py

```diff
@@ -89,6 +89,16 @@
 # This patterns also effect to html_static_path and html_extra_path
 exclude_patterns = []
 
+# List of warnings to suppress
+suppress_warnings = []
+
+# if the file deprecated.xml does not exist in the doxygen xml output, breathe
+# will fail to build the docs, so we conditionally add "breathe.rst" to the
+# exclude_patterns list
+if not os.path.exists(breathe_projects["librmm"] + "/deprecated.xml"):
+    exclude_patterns.append("librmm_docs/deprecated.rst")
+    suppress_warnings.append("toc.excluded")
```

Comment:
What is this for? I'd guess when you exclude things it generates an extra file and then it warns about that file not being included anywhere, or something like that?

Reply:
Because `deprecated` is listed in the toc in the main RST file. This lets us leave it there deprecated.rst file is excluded. Without this suppression you get a warning that an excluded file is in the toc.

Reply:
I didn't have a better way to conditionally remove it from the toc.

Reply:
Ah OK I see. Yeah that seems fine then.